### PR TITLE
doc: update documentation with v2.0.0.rc0 changes

### PR DESCRIPTION
**Pull request by laurenyu**
*Issue #, if available:*

*Description of changes:*
doc update in preparation of releasing v2.0.0.rc0. I know we typically don't update the changelog manually (anymore), but since the release will be GitHub-only and not on PyPI, I don't plan on utilizing our CI/CD for it.

*Testing done:*
built the docs locally

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to any/all clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/v2.rst [hunk 1]
```diff
@@ -35,7 +35,88 @@ If you are executing this pip install command in a notebook, make sure to restar
 Changes
 *******
 
-To see what changes have been made, see the `CHANGELOG <https://github.com/aws/sagemaker-python-sdk/blob/zwei/CHANGELOG.md>`_.
+This section is for major changes that may require updates to your SageMaker Python SDK code.
+You can also see what changes have been made in the `CHANGELOG <https://github.com/aws/sagemaker-python-sdk/blob/zwei/CHANGELOG.md>`_.
```
Comment (aaronmarkham) [inline doc/v2.rst:39]: Why in the zwei branch?

Comment (aaronmarkham) [reply doc/v2.rst:39]: I see... this is going into that branch...

Comment (laurenyu) [reply doc/v2.rst:39]: yeah. everything will get merged into the main branch when v2.0 is finally released

**Comment by metrizable** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-python-sdk-unit-tests
* Commit ID: 9534b321830980a2669153366598b8f29b799345
* Result: SUCCEEDED
* [Build Logs](https://t550wuxq34.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=11b5640a-a173-49ef-a932-f3fdd033344c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**